### ISSUE TITLE
ambiguous options.conf: allow_test_failures = false

### ISSUE THREAD

**Issue by jurobystricky** [clearlinux/autospec#467]
If I specify in options.conf:
**allow_test_failures = false**

Then, as expected,  the spec file contains something like:
<pre>
make VERBOSE=1 V=1 %{?_smp_mflags} check
</pre>

However, when avx2 is enabled, "**use_avx2 = true**"
the spec file adds:
<pre>
cd ../buildavx2;
make VERBOSE=1 V=1 %{?_smp_mflags} check || :
</pre>
So for avx2 (and avx512) the failures are allowed in spite of **allow_test_failures = false.**
This is probably by design in order to allow running autospec on machines without avx2/avx512 hardware support. Nevertheless, there are valid cases when you do want to run avx2/avx512 tests and fail the build if check fails. So autospec could detect avx presence and issue a warning if building on machine without sufficient support (and adjust the spec file).

**Comment by phmccarty** [comment]
@jurobystricky I agree that test suites should not fail unconditionally in this case.

One option would be to check host system capabilities prior to running the `make check` command and adjust error handling as needed. In pseudocode:

```
if host supports avx512:
  run command1
else if host supports avx2:
  run command2
else:
  run command3
```
